sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -1548,14 +1548,27 @@ public Node visitWhenClause(SqlBaseParser.WhenClauseContext context) {
 
     @Override
     public Node visitFunctionCall(SqlBaseParser.FunctionCallContext context) {
+        var window = visitIfPresent(context.over(), Window.class);
+        var filter = visitIfPresent(context.filter(), Expression.class);
+        if (filter.isPresent() && window.isPresent()) {
+            throw new UnsupportedOperationException(
+                "The filter clause in window function calls is not supported");
```

Comment:
Not sure if this is a good place for the current limitation for window functions. The same limitation applies for non-aggregate functions so we'd have to limit that elsewhere anyway. I'd remove this limitation at the parser level.